.github/workflows/build.yaml

```diff
@@ -46,10 +64,51 @@ jobs:
         PYTHON: python
     - name: Install dependencies
       run: make setup
+    - name: Create changelog and bump
+      if: ${{ github.ref == 'refs/heads/main' && matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON }}
+      run: |
+        git config --global user.name "github-actions"
+        git config --global user.email "github-actions@github.com"
+        git config --list --global
+        # First generate the CHANGELOG diff to use as release notes.
+        cz changelog --dry-run --incremental --unreleased-version "Release Notes" > ${{ env.ARTIFACT_DIR }}/RELEASE_NOTES.md
+        # Update the CHANGELOG.md.
+        cz changelog
+        cz bump
     - name: Build the package
-      # We don't need to check and test the package separately
-      # because `make dist` runs those targets first and only
-      # builds the package if they succeed.
-      run: make dist
-    - name: Build the docs
-      run: make docs
+      # We don't need to check and test the package separately because `make dist` runs
+      # those targets first and only builds the package if they succeed.
+      run: |
+        make dist requirements sbom
+    - name: Compute package hash
+      if: ${{ matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON }}
+      id: compute-hash
+      shell: bash
+      run: |
+        set -euo pipefail
+        TARBALL_PATH=$(find ${{ env.ARTIFACT_DIR }} -name "*.tar.gz")
+        WHEEL_PATH=$(find ${{ env.ARTIFACT_DIR }} -name "*.whl")
+        SBOM_PATH=$(find ${{ env.ARTIFACT_DIR }} -name "*-sbom.json")
+        HTML_DOCS_PATH=$(find ${{ env.ARTIFACT_DIR }} -name *-docs-html.zip)
```

Comment:
@behnazh with https://github.com/jenstroeger/python-package-template/pull/263#discussion_r923930333 in mind, couldn’t we use
```
TARBALL_PATH=$(find $ARTIFACT_DIR -name "*.tar.gz")
```
here instead? And anywhere else where we use `${{ env.BLA }}` inside of a shell 🤔

Reply:
Thanks. Yes the env variables in bash don't need the `env` [context](https://docs.github.com/en/actions/learn-github-actions/environment-variables#using-contexts-to-access-environment-variable-values). However, we should be careful about Windows OS because the syntax for PowerShell is `$env:NAME`.

Reply:
> However, we should be careful about Windows OS because the syntax for PowerShell is `$env:NAME`.

Hmm, I wonder if we can set the shell for steps across platforms, e.g. using [`defaults.run`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaultsrun)—would that allow us to use bash on Windows runners as well?